src/components/Choropleth.vue

```diff
@@ -142,20 +145,48 @@ const style = (f: Feature) => {
 
 const getLeafletMap = () => map.value?.leafletObject;
 
-const updateBounds = async () => {
+const lockBounds = async () => {
     const leafletMap = getLeafletMap();
     if (!leafletMap) return;
 
-    const country = selectedCountryId.value || "GLOBAL";
-    const [west, east, south, north] = countryBoundingBoxes.value[country];
-    const countryBounds = new LatLngBounds([south, west], [north, east]);
-    await leafletMap.fitBounds(countryBounds);
+    if (isNewSelectedCountry.value && selectedCountryId.value) {
+        leafletMap.setMaxBounds(leafletMap.getBounds());
+        leafletMap.setMinZoom(leafletMap.getZoom());
+    }
 
-    // record bounds for testing
-    bounds.value = { west, east, south, north };
+    isNewSelectedCountry.value = false;
+    waitingForMapBounds.value = false;
 };
 
-const updateMap = async (newFeatures: Feature[]) => {
+const getBoundsForCountry = (countryId?: string) => {
```

Comment:
```
const getBounds = (countryId?: string = "GLOBAL") => {
```
Maybe rename because this function doesn't always deal with countries. `getBounds` or `getRegionBounds`?

Reply:
I'd vote for `getRegionBounds`

Reply:
done